packages/notifier/test/test-notifier-adaptor.js

```diff
@@ -42,7 +43,8 @@ const finiteStream = makeIterable(false);
 const explodingStream = makeIterable(true);
 
 const testEnding = (t, p, fails) => {
-  return Promise.resolve(p).then(
+  return E.when(
```

Comment:
Something about this change is failing tests.  Can you inspect the `E.when` implementation?

Reply:
Then `when` implementation is fine, as is the line above. Rather, the problem is that the old `when` followed by `then` was causing an extra turn, leading to incorrect "lossy" testing happening to succeed. With only one `when` the order changed, the notifier lossiness changed, exposing a pattern of bugs in my lossiness testing. Fixed.